node/clients/profile/profileV2.ts

```diff
@@ -0,0 +1,303 @@
+import {
+  InstanceOptions,
+  IOContext,
+  ExternalClient,
+  RequestConfig,
+} from '@vtex/api'
+import { AxiosError } from 'axios'
+
+import { statusToError } from '../../utils'
+
+export class ProfileClientV2 extends ExternalClient {
+  account: string
+  defaultPIIRequest: PIIRequest
+
+  constructor(baseUrl: string, context: IOContext, options?: InstanceOptions) {
+    super(baseUrl, context, options)
+
+    this.account = context.account
+
+    this.defaultPIIRequest = {
+      useCase: "MyAcocunts",
+      onBehalfOf: "user"
+    } as PIIRequest
+  }
+
+  public getProfileInfo = (user: CurrentProfile, customFields?: string, piiRequest?: PIIRequest) => {
+    let { userKey, alternativeKey } = this.getUserKeyAndAlternateKey(user)
+    let url = this.getPIIUrl(`${this.baseUrl}/${userKey}`, alternativeKey, piiRequest)
+
+    return this.get<ProfileV2>(
+      url,
+      {
+        metric: 'profile-system-v2-getProfileInfo',
+        params: {
+          extraFields: customFields,
+        },
+      }
+    ).then((profile: ProfileV2[]) => {
+      if (profile.length > 0) {
+        let profileV2 = profile[0].document
+        profileV2.pii = true
+        profileV2.id = profile[0].id
+        profileV2.isNewsletterOptIn = profileV2.isNewsletterOptIn || false
+
+        return profileV2
+      }
+
+      return {} as Profile
+    })
+  }
+
+  public createProfile = (profile: Profile) =>
+    this.post<ProfileV2>(`${this.baseUrl}?an=${this.account}`, profile)
+      .then((profileV2: ProfileV2) => {
+        let newProfile = profileV2.document as Profile
+        newProfile.id = profileV2.id
+        return newProfile
+      })
+
+  public updatePersonalPreferences = (
+    user: CurrentProfile,
+    personalPreferences: PersonalPreferences
+  ) => {
+    let { userKey, alternativeKey } = this.getUserKeyAndAlternateKey(user)
+    let parsedPersonalPreferences = Object.keys(personalPreferences).reduce((preferences: any, key) => {
+      preferences[key] = personalPreferences[key] === 'True'
+      return preferences
+    }, {})
```

Comment:
```suggestion
    const parsedPersonalPreferences = Object.fromEntries(Object.entries(personalPreferences).map(([key, value]) => {
      return [key, value === 'True']
    }))
```
To avoid mutability, reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries

Reply:
Thanks!